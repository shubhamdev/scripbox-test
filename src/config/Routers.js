import React from "react";
import { Link, Switch } from "react-router-dom";
import PrivateRoute from "./PrivateRoutes";
import PublicRoute from "./PublicRoutes";
import { Layout } from "antd";
const { Header, Content } = Layout;
const LoginPage = React.lazy(() => import("../components/loginPage/LoginPage"));
const ChallengesPage = React.lazy(() =>
  import("../components/challengesPage/ChallengesPage")
);
const DashboardPage = React.lazy(() =>
  import("../components/dashboardPage/DashboardPage")
);

const HomePage = React.lazy(() => import("../components/homePage/HomePage"));
function NoMatch() {
  return (
    <div
      style={{
        alignItems: "center",
        display: "flex",
        justifyContent: "center",
        padding: "1rem",
        height: "90vh",
      }}
    >
      <h3>
        No match for this route
        <Link to="/"> Login</Link>
      </h3>
    </div>
  );
}

function Routers() {
  return (
    <Layout>
      <Header>
        <HomePage />
      </Header>
      <Content style={{ height: "100bh" }}>
        <Switch>
          <PublicRoute
            restricted={false}
            component={LoginPage}
            path="/"
            exact
          />
          <PrivateRoute component={ChallengesPage} path="/challenge" exact />
          <PrivateRoute component={DashboardPage} path="/dashboard" exact />
          <PublicRoute path="*">
            <NoMatch />
          </PublicRoute>
        </Switch>
      </Content>
    </Layout>
  );
}

export default Routers;
