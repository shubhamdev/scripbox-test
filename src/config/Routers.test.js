import React from "react";
import { unmountComponentAtNode } from "react-dom";
import { shallow } from "enzyme";
import Routers from "./Routers";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("Routers", () => {
  it('should render Routers correctly in "debug" mode', () => {
    const component = shallow(<Routers debug />);
    expect(component).toMatchSnapshot();
  });
  it("should render Routers correctly with no props", () => {
    const component = shallow(<Routers />);
    expect(component).toMatchSnapshot();
  });
});
