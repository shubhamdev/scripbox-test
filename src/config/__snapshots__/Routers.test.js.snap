// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routers should render Routers correctly in "debug" mode 1`] = `
<Layout>
  <Header>
    <lazy />
  </Header>
  <Content
    style={
      Object {
        "height": "100bh",
      }
    }
  >
    <Switch>
      <PublicRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/"
        restricted={false}
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/challenge"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/dashboard"
      />
      <PublicRoute
        path="*"
      >
        <NoMatch />
      </PublicRoute>
    </Switch>
  </Content>
</Layout>
`;

exports[`Routers should render Routers correctly with no props 1`] = `
<Layout>
  <Header>
    <lazy />
  </Header>
  <Content
    style={
      Object {
        "height": "100bh",
      }
    }
  >
    <Switch>
      <PublicRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/"
        restricted={false}
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/challenge"
      />
      <PrivateRoute
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_init": [Function],
            "_payload": Object {
              "_result": [Function],
              "_status": -1,
            },
          }
        }
        exact={true}
        path="/dashboard"
      />
      <PublicRoute
        path="*"
      >
        <NoMatch />
      </PublicRoute>
    </Switch>
  </Content>
</Layout>
`;
